lib/VolumeManager/LVM/lvm2disk.rb

```diff
@@ -42,7 +42,12 @@ def d_read(pos, len)
     if logicalVolume.thin?
       device_id = logicalVolume.thin_segment.device_id
       thin_pool = logicalVolume.thin_pool_volume
-      data_blks = thin_pool.metadata_volume.superblock.device_to_data(device_id, pos, len)
+      begin
+        data_blks = thin_pool.metadata_volume.superblock.device_to_data(device_id, pos, len)
+      rescue
+        return retStr
```

Comment:
@movitto Should it return an empty string, or a string of 0x0 of the requested length?
That is, unless the length causes the read to exceed the end of the device, then it should return a string of 0x0
of the available length.

Reads from a block device should only return an empty string when at the end of the device. I would think LVM2 would maintain that behavior.